apps/entry/views.py

```diff
@@ -10,13 +11,35 @@
 from rest_framework.response import Response
 from django.shortcuts import redirect
 
+from apps.common.utils import (
+    EXTERNAL_TUPLE_SEPARATOR,
+    EXTERNAL_ARRAY_SEPARATOR,
+    EXTERNAL_FIELD_SEPARATOR,
+    format_locations,
+)
 from apps.gidd.views import client_id
 from utils.common import track_gidd
-from apps.common.utils import EXTERNAL_TUPLE_SEPARATOR
+from utils.db import Array
 
 
-def get_idu_data():
-    return Figure.objects.annotate(
+def extract_location_data(data):
+    # Split the formatted location data into individual components
+    location_components = [
+        loc.split(EXTERNAL_FIELD_SEPARATOR)
+        for loc in format_locations(data).split(EXTERNAL_ARRAY_SEPARATOR)
+    ]
```

Comment:
Just a Note:
After @Rup-Narayan-Rajbanshi locations work is merged, we don't need to use split anymore :D
```suggestion
    location_components = format_locations(data)
```